# **Análise de Condições para ocorrência de Alzhaimer**
dataset do kaggle disponível em: https://www.kaggle.com/datasets/rabieelkharoua/alzheimers-disease-dataset

# 0 Importações

## 0.1 Importação das bibliotecas

In [35]:
#Bibliotecas de manipulação de dados
import pandas as pd

#Bibliotecas de cálculos
import numpy as np
import math
from sklearn.metrics import confusion_matrix

#Bibliotecas de visualização de dados
import seaborn as sns
import matplotlib.pyplot as plt
import plotly.express as px
import plotly.graph_objects as go
from plotly.subplots import make_subplots

#Utilidades
import inflection

## 0.2 Carregamento dos dados

In [11]:
df_raw = pd.read_csv("alzheimers_disease_data.csv")
df_raw.head()

,PatientID,Age,Gender,Ethnicity,EducationLevel,BMI,Smoking,AlcoholConsumption,PhysicalActivity,DietQuality,...,MemoryComplaints,BehavioralProblems,ADL,Confusion,Disorientation,PersonalityChanges,DifficultyCompletingTasks,Forgetfulness,Diagnosis,DoctorInCharge
0,4751,73,0,0,2,22.927749,0,13.297218,6.327112,1.347214,...,0,0,1.725883,0,0,0,1,0,0,XXXConfid
1,4752,89,0,0,0,26.827681,0,4.542524,7.619885,0.518767,...,0,0,2.592424,0,0,0,0,1,0,XXXConfid
2,4753,73,0,3,1,17.795882,0,19.555085,7.844988,1.826335,...,0,0,7.119548,0,1,0,1,0,0,XXXConfid
3,4754,74,1,0,1,33.800817,1,12.209266,8.428001,7.435604,...,0,1,6.481226,0,0,0,0,0,0,XXXConfid
4,4755,89,0,0,0,20.716974,0,18.454356,6.310461,0.795498,...,0,0,0.014691,0,0,1,1,0,0,XXXConfid


# 1.0 Descrição dos dados

In [63]:
df1 = df_raw.copy()

## Descrição das colunas

- **Tabela de Conteúdo**  
- **Informações do Paciente**  
- **ID do Paciente**  
- **Detalhes Demográficos**  
- **Fatores de Estilo de Vida**  
- **Histórico Médico**  
- **Medições Clínicas**  
- **Avaliações Cognitivas e Funcionais**  
- **Sintomas**  
- **Informações de Diagnóstico**  
- **Informações Confidenciais**  

### **Informações do Paciente**  
- **PatientID**: Identificador único atribuído a cada paciente (4751 a 6900).  

### **Detalhes Demográficos**  
- **Age**: Idade dos pacientes, variando de 60 a 90 anos.  
- **Gender**: Gênero dos pacientes, onde 0 representa Masculino e 1 representa Feminino.  
- **Ethnicity**: Etnia dos pacientes, codificada da seguinte forma:  
  - 0: Caucasiano  
  - 1: Afro-Americano  
  - 2: Asiático  
  - 3: Outro  
- **EducationLevel**: Nível de educação dos pacientes, codificado da seguinte forma:  
  - 0: Nenhum  
  - 1: Ensino Médio  
  - 2: Bacharelado  
  - 3: Superior  

### **Fatores de Estilo de Vida**  
- **BMI**: Índice de Massa Corporal dos pacientes, variando de 15 a 40.  
- **Smoking**: Status de tabagismo, onde 0 indica Não e 1 indica Sim.  
- **AlcoholConsumption**: Consumo semanal de álcool em unidades, variando de 0 a 20.  
- **PhysicalActivity**: Atividade física semanal em horas, variando de 0 a 10.  
- **DietQuality**: Pontuação da qualidade da dieta, variando de 0 a 10.  
- **SleepQuality**: Pontuação da qualidade do sono, variando de 4 a 10.  

### **Histórico Médico**  
- **FamilyHistoryAlzheimers**: Histórico familiar de Doença de Alzheimer, onde 0 indica Não e 1 indica Sim.  
- **CardiovascularDisease**: Presença de doença cardiovascular, onde 0 indica Não e 1 indica Sim.  
- **Diabetes**: Presença de diabetes, onde 0 indica Não e 1 indica Sim.  
- **Depression**: Presença de depressão, onde 0 indica Não e 1 indica Sim.  
- **HeadInjury**: Histórico de lesão na cabeça, onde 0 indica Não e 1 indica Sim.  
- **Hypertension**: Presença de hipertensão, onde 0 indica Não e 1 indica Sim.  

### **Medições Clínicas**  
- **SystolicBP**: Pressão arterial sistólica, variando de 90 a 180 mmHg.  
- **DiastolicBP**: Pressão arterial diastólica, variando de 60 a 120 mmHg.  
- **CholesterolTotal**: Níveis totais de colesterol, variando de 150 a 300 mg/dL.  
- **CholesterolLDL**: Níveis de colesterol de lipoproteína de baixa densidade (LDL), variando de 50 a 200 mg/dL.  
- **CholesterolHDL**: Níveis de colesterol de lipoproteína de alta densidade (HDL), variando de 20 a 100 mg/dL.  
- **CholesterolTriglycerides**: Níveis de triglicerídeos, variando de 50 a 400 mg/dL.  

### **Avaliações Cognitivas e Funcionais**  
- **MMSE**: Pontuação do Mini Exame do Estado Mental, variando de 0 a 30. Pontuações mais baixas indicam comprometimento cognitivo.  
- **FunctionalAssessment**: Pontuação de avaliação funcional, variando de 0 a 10. Pontuações mais baixas indicam maior comprometimento.  
- **MemoryComplaints**: Presença de queixas de memória, onde 0 indica Não e 1 indica Sim.  
- **BehavioralProblems**: Presença de problemas comportamentais, onde 0 indica Não e 1 indica Sim.  
- **ADL**: Pontuação de atividades da vida diária, variando de 0 a 10. Pontuações mais baixas indicam maior comprometimento.  

### **Sintomas**  
- **Confusion**: Presença de confusão, onde 0 indica Não e 1 indica Sim.  
- **Disorientation**: Presença de desorientação, onde 0 indica Não e 1 indica Sim.  
- **PersonalityChanges**: Presença de mudanças de personalidade, onde 0 indica Não e 1 indica Sim.  
- **DifficultyCompletingTasks**: Presença de dificuldade em completar tarefas, onde 0 indica Não e 1 indica Sim.  
- **Forgetfulness**: Presença de esquecimento, onde 0 indica Não e 1 indica Sim.  

### **Informações de Diagnóstico**  
- **Diagnosis**: Status de diagnóstico da Doença de Alzheimer, onde 0 indica Não e 1 indica Sim.  

### **Informações Confidenciais**  
- **DoctorInCharge**: Esta coluna contém informações confidenciais sobre o médico responsável, com o valor "XXXConfid" para todos os pacientes.  

## 1.1 Renomeando colunas

In [65]:
colunas = df1.columns

snakecase = lambda x: inflection.underscore( x )

colunas_corrigidas = list(map(snakecase, colunas))

df1.columns = colunas_corrigidas
df1.columns

Index(['patient_id', 'age', 'gender', 'ethnicity', 'education_level', 'bmi',
       'smoking', 'alcohol_consumption', 'physical_activity', 'diet_quality',
       'sleep_quality', 'family_history_alzheimers', 'cardiovascular_disease',
       'diabetes', 'depression', 'head_injury', 'hypertension', 'systolic_bp',
       'diastolic_bp', 'cholesterol_total', 'cholesterol_ldl',
       'cholesterol_hdl', 'cholesterol_triglycerides', 'mmse',
       'functional_assessment', 'memory_complaints', 'behavioral_problems',
       'adl', 'confusion', 'disorientation', 'personality_changes',
       'difficulty_completing_tasks', 'forgetfulness', 'diagnosis',
       'doctor_in_charge'],
      dtype='object')

## 1.2 Dimensão dos dados

In [42]:
print(f"Linhas: {df1.shape[0]}\nColunas: {df1.shape[1]}")

Linhas: 2149
Colunas: 35


## 1.3 Tipo das colunas

In [45]:
df1.dtypes

patient_id                       int64
age                              int64
gender                           int64
ethnicity                        int64
education_level                  int64
bmi                            float64
smoking                          int64
alcohol_consumption            float64
physical_activity              float64
diet_quality                   float64
sleep_quality                  float64
family_history_alzheimers        int64
cardiovascular_disease           int64
diabetes                         int64
depression                       int64
head_injury                      int64
hypertension                     int64
systolic_bp                      int64
diastolic_bp                     int64
cholesterol_total              float64
cholesterol_ldl                float64
cholesterol_hdl                float64
cholesterol_triglycerides      float64
mmse                           float64
functional_assessment          float64
memory_complaints        

### 1.3.1 Ajustando as colunas

In [67]:
replace_gender = {0: "male", 1: "female"}
df1["gender"] = df1["gender"].map(replace_gender)

replace_ethnicity = {0: "caucasian", 1: "african american", 2: "asian", 3: "other"}
df1["ethnicity"] = df1["ethnicity"].map(replace_ethnicity)

replace_educational = {0: "none", 1: "high school", 2: "bachelor's", 3: "Higher"}
df1["education_level"] = df1["education_level"].map(replace_educational)

replace_binary = {0: "no", 1: "yes"}
cols_binary = ["smoking", "family_history_alzheimers", "cardiovascular_disease", "diabetes", "depression", "head_injury", "hypertension", "memory_complaints", "behavioral_problems", "confusion", "disorientation", "personality_changes", "difficulty_completing_tasks", "forgetfulness", "diagnosis"]

for col in cols_binary:
    df1[col] = df1[col].map(replace_binary)

In [69]:
df1.head()

,patient_id,age,gender,ethnicity,education_level,bmi,smoking,alcohol_consumption,physical_activity,diet_quality,...,memory_complaints,behavioral_problems,adl,confusion,disorientation,personality_changes,difficulty_completing_tasks,forgetfulness,diagnosis,doctor_in_charge
0,4751,73,male,caucasian,bachelor's,22.927749,no,13.297218,6.327112,1.347214,...,no,no,1.725883,no,no,no,yes,no,no,XXXConfid
1,4752,89,male,caucasian,none,26.827681,no,4.542524,7.619885,0.518767,...,no,no,2.592424,no,no,no,no,yes,no,XXXConfid
2,4753,73,male,other,high school,17.795882,no,19.555085,7.844988,1.826335,...,no,no,7.119548,no,yes,no,yes,no,no,XXXConfid
3,4754,74,female,caucasian,high school,33.800817,yes,12.209266,8.428001,7.435604,...,no,yes,6.481226,no,no,no,no,no,no,XXXConfid
4,4755,89,male,caucasian,none,20.716974,no,18.454356,6.310461,0.795498,...,no,no,0.014691,no,no,yes,yes,no,no,XXXConfid


## 1.4 Checagem de NAs

In [71]:
df1.isna().sum()

patient_id                     0
age                            0
gender                         0
ethnicity                      0
education_level                0
bmi                            0
smoking                        0
alcohol_consumption            0
physical_activity              0
diet_quality                   0
sleep_quality                  0
family_history_alzheimers      0
cardiovascular_disease         0
diabetes                       0
depression                     0
head_injury                    0
hypertension                   0
systolic_bp                    0
diastolic_bp                   0
cholesterol_total              0
cholesterol_ldl                0
cholesterol_hdl                0
cholesterol_triglycerides      0
mmse                           0
functional_assessment          0
memory_complaints              0
behavioral_problems            0
adl                            0
confusion                      0
disorientation                 0
personalit

## 1.5 Descrição dos dados

In [73]:
df_num = df1.select_dtypes( include=["int64", "float64"] )
df_cat = df1.select_dtypes( exclude=["int64", "float64"] )

### 1.5.1 Dados numéricos

In [76]:
medias = pd.DataFrame( df_num.apply( np.mean ) ).T 
medianas = pd.DataFrame( df_num.apply( np.median ) ).T 

In [78]:
desvio_padrao = pd.DataFrame( df_num.apply( np.std ) ).T
minimo = pd.DataFrame( df_num.apply( min ) ).T
maximo = pd.DataFrame( df_num.apply( max ) ).T
intervalo = pd.DataFrame( df_num.apply(lambda x: x.max() - x.min()) ).T
skew = pd.DataFrame( df_num.apply( lambda x: x.skew())).T 
kurtosis = pd.DataFrame( df_num.apply(lambda x: x.kurtosis() ) ).T

In [80]:
metricas_numericas = pd.concat( [minimo, maximo, intervalo, medias, medianas, desvio_padrao, skew, kurtosis] ).T.reset_index()
metricas_numericas.columns = (["atributos","minimo","maximo","intervalo","medias","medianas","desvio_padrao","skew","kurtosis"])

In [82]:
metricas_numericas

,atributos,minimo,maximo,intervalo,medias,medianas,desvio_padrao,skew,kurtosis
0,patient_id,4751.000000,6899.000000,2148.000000,5825.000000,5825.000000,620.362797,0.000000,-1.200000
1,age,60.000000,90.000000,30.000000,74.908795,75.000000,8.988129,0.045964,-1.189214
2,bmi,15.008851,39.992767,24.983916,27.655697,27.823924,7.215759,-0.026715,-1.184508
3,alcohol_consumption,0.002003,19.989293,19.987290,10.039442,9.934412,5.756570,0.018415,-1.202876
4,physical_activity,0.003616,9.987429,9.983813,4.920202,4.766424,2.856526,0.044973,-1.179196
5,diet_quality,0.009385,9.998346,9.988961,4.993138,5.076087,2.908378,-0.012058,-1.228962
6,sleep_quality,4.002629,9.999840,5.997212,7.051081,7.115646,1.763163,-0.069630,-1.212454
7,systolic_bp,90.000000,179.000000,89.000000,134.264774,134.000000,25.943314,0.009971,-1.197593
8,diastolic_bp,60.000000,119.000000,59.000000,89.847836,91.000000,17.588403,-0.054470,-1.235040
9,cholesterol_total,150.093316,299.993352,149.900037,225.197519,225.086430,42.532334,-0.018674,-1.156042


### 1.5.2 Dados categóricos

In [84]:
df_cat.apply(lambda x: x.unique().shape[0])

gender                         2
ethnicity                      4
education_level                4
smoking                        2
family_history_alzheimers      2
cardiovascular_disease         2
diabetes                       2
depression                     2
head_injury                    2
hypertension                   2
memory_complaints              2
behavioral_problems            2
confusion                      2
disorientation                 2
personality_changes            2
difficulty_completing_tasks    2
forgetfulness                  2
diagnosis                      2
doctor_in_charge               1
dtype: int64

# 2.0 Filtragem de variáveis

In [91]:
df2 = df1.copy()

In [93]:
#Remoção de colunas de ID e de valores únicos
cols_drop = ["patient_id", "doctor_in_charge"]
df2.drop(cols_drop, axis=1, inplace=True)

# 3.0 Analise exploratória

In [31]:
df3 = df1.copy()

## 3.1 Análise univariada

### 3.1.1 Análise das variáveis numéricas

### 2.1.2 Análise das variáveis categoricas

## 2.2 Análise Bivariada

### 2.2.1 Análise das variáveis numéricas

### 2.2.1 Análise visando a ocorrência de AVC